lxd/instance/instance_utils.go

```diff
@@ -112,6 +112,10 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, expanded bool, instanc
 		return fmt.Errorf("nvidia.runtime is incompatible with privileged containers")
 	}
 
+	if sysOS.InUbuntuCore() && shared.IsTrue(config["nvidia.runtime"]) {
+		return fmt.Errorf("nvidia.runtime is incompatible with Ubuntu Core")
```

Comment:
@mihalicyn is nvidia.runtime compatible with CDI mode in general? I'm wondering if we need this check specifically (although happy to keep it if CDI does work with nvidia.runtime).

Reply:
As far as I understand `nvidia.runtime=true` thing and CDI thing are absolute independent from each other. `nvidia.runtime=true` uses https://github.com/NVIDIA/libnvidia-container and https://github.com/lxc/lxc/blob/main/hooks/nvidia hook but I wasn't able to make it work (and stopped trying as we decided to go with a CDI thing, i.e. https://github.com/NVIDIA/nvidia-container-toolkit).

I guess this check is really useful as I'm pretty sure that Ubuntu Core users will try to set `nvidia.runtime=true` and it will silently fail and all NVIDIA tooling won't be available inside the container. IMHO, it's better to explicitly say that it does not supported at all.

Reply:
i agree, but should we also refuse to support it for CDI mode too?

Reply:
@gabrielmougard should nvidia.runtime be blocked from being set when using CDI mode?

Reply:
> @gabrielmougard should nvidia.runtime be blocked from being set when using CDI mode?

Yes I think so. As @mihalicyn said, the CDI approach and the LXC hook called if `nvidia.runtime=true` are independent. Plus, having both option set together could potentially introduce linking issues: both approaches update the container `ldcache`, so if libnvidia (used by the legacy NVIDIA hook for LXC) hasn't the same library lookup mechanisn than nvctk (CDI), it could override the ldcache config and break things inside the container that will only be seen at runtime (when executing a CUDA workload for example (something like `libnvml not found` for example))

Reply:
It's safer to restrict the user from setting CDI mode and nvidia.runtime=true ... It'll avoid these nasty runtime bugs.

Reply:
Thanks!

Let me add some sanity logic around that then.

Reply:
Thanks @gabrielmougard 
And im happy to keep the existing check for ubuntu core and nvidia runtime too @mihalicyn